File: Task/Averages-Mode/Go/averages-mode-2.go
package main

import "fmt"

func main() {
    fmt.Println(mode([]interface{}{.2, .7, .1, .8, .2}))
    fmt.Println(mode([]interface{}{"two", 7, 1, 8, "two", 8}))
}

func mode(a []interface{}) []interface{} {
    m := make(map[interface{}]int)
    for _, v := range a {
        m[v]++
    }
    var mode []interface{}
    var n int
    for k, v := range m {
        switch {
        case v < n:
        case v > n:
            n = v
            mode = append(mode[:0], k)
        default:
            mode = append(mode, k)
        }
    }
    return mode
}

File: Task/Averages-Mode/F-Sharp/averages-mode-2.fs
> mode ["a"; "b"; "c"; "c"];;
val it : string list = ["c"]
> mode ["a"; "b"; "c"; "c";"a"];;
val it : string list = ["c"; "a"]
> mode [1;2;1;3;2;0;0];;
val it : int list = [0; 2; 1]
